src/chi/javascript/components/modal.js

```diff
@@ -0,0 +1,232 @@
+              Util.removeClass(self._backdrop, chi.classes.CLOSED);
+            }
+            self._shown = true;
+            self._modalElem.dispatchEvent(
+              Util.createEvent(EVENTS.show)
+            );
+          },
+          function() {
+            Util.removeClass(self._modalElem, chi.classes.TRANSITIONING);
+            if (self._backdrop) {
+              Util.removeClass(self._backdrop, chi.classes.TRANSITIONING);
+            }
+            self._transitioning = false;
+          },
+          ANIMATION_DURATION
+        );
+      } else {
+        Util.addClass(self._modalElem, chi.classes.ACTIVE);
+        Util.addClass(self._elem, chi.classes.ACTIVE);
+        if (self._backdrop) { Util.removeClass(self._backdrop, chi.classes.CLOSED); }
+        self._transitioning = false;
+        self._shown = true;
+        self._modalElem.dispatchEvent(
+          Util.createEvent(EVENTS.show)
+        );
+      }
+    }
+  }
+
+  hide() {
+    if (this._shown) {
+      if (this._transitioning) {
+        Util.stopThreeStepsAnimation(this._currentThreeStepsAnimation, false);
+      }
+      this._transitioning = true;
+      const self = this;
+      if (this._config.animated){
+        this._currentThreeStepsAnimation = Util.threeStepsAnimation(
+          function() {
+            Util.addClass(self._modalElem, chi.classes.TRANSITIONING);
+            if (self._backdrop) {
+              Util.addClass(self._backdrop, chi.classes.TRANSITIONING);
+            }
+          },
+          function(){
+            Util.removeClass(self._modalElem, chi.classes.ACTIVE);
+            Util.removeClass(self._elem, chi.classes.ACTIVE);
+            if (self._backdrop) {
+              Util.addClass(self._backdrop, chi.classes.CLOSED);
+            }
+            self._shown = false;
+            self._modalElem.dispatchEvent(
+              Util.createEvent(EVENTS.hide)
+            );
+          },
+          function() {
+            Util.removeClass(self._modalElem, chi.classes.TRANSITIONING);
+            if (self._backdrop) {
+              Util.removeClass(self._backdrop, chi.classes.TRANSITIONING);
+            }
+            self._transitioning = false;
+          },
+          ANIMATION_DURATION
+        );
+      } else {
+        Util.removeClass(self._modalElem, chi.classes.ACTIVE);
+        Util.removeClass(self._elem, chi.classes.ACTIVE);
+        if (self._backdrop) { Util.addClass(self._backdrop, chi.classes.CLOSED); }
+        self._transitioning = false;
+        self._shown = false;
+        self._modalElem.dispatchEvent(
+          Util.createEvent(EVENTS.hide)
+        );
+      }
+    }
+  }
+
+  toggle () {
+    if (this._shown) {
+      this.hide();
+    } else {
+      this.show();
+    }
+  }
+
+  dispose() {
+    this._config = null;
+    this._modalElem = null;
+    this._shown = null;
+    this._transitioning = null;
+    this._elem.removeEventListener('click', this._triggerClickEventListener);
```

Comment:
Keyup event listener should be removed from document.body. You should use method this._addEventHandler from the Component class (Modal inherits this class) to add all event listeners and then you'll only need to call this._removeEventHandlers() to remove all at once. This ensures you don't forget to remove any event listener.